detection_rules/rule.py

```diff
@@ -689,7 +723,7 @@ def _add_required_fields(self, obj: dict) -> None:
             required_fields = []
 
         field_name = "required_fields"
-        if self.check_restricted_field_version(field_name=field_name):
+        if required_fields and self.check_restricted_field_version(field_name=field_name):
```

Comment:
Similar to [related_integrations](https://github.com/elastic/detection-rules/pull/2060/files#r938282590), I updated this so that `required_fields` _only_ adds the value if set to avoid empty arrays